src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -684,7 +684,7 @@ internal override (CSharpAttributeData?, BoundAttribute?) EarlyDecodeWellKnownAt
                         Debug.Assert(constructorArguments.Length == 1);
                         if (constructorArguments[0].TryDecodeValue(SpecialType.System_String, out string? parameterName))
                         {
-                            var parameters = ContainingSymbol.GetParameters();
+                            var parameters = ContainingSymbol.GetParametersIncludingExtensionParameter(skipExtensionIfStatic: true);
```

Comment:
>GetParametersIncludingExtensionParameter [](http://example.com/codeflow?start=62&length=40)

I think the ```index``` should not depend on whether we were able to create a symbol for the receiver parameter. In other words, `0` should always be reserved for a receiver when we are dealing with non-static extension member. If the receiver parameter is not available, that just means that the attribute won't match it and the ```index``` won't be `0`. It looks like usage of this helper won't give us this behavior